### PR TITLE
feat: use notifyAll insteadOf notify for listener events notify

### PR DESCRIPTION
**Pull request by thehackercat**
Signed-off-by: xiaoxu <lexuscyborg103@gmail.com>

**What this PR does**: 
image

`Notify one after one` might have performance issues, this PR makes serviceListeners notify events concurrently and do not wait for notification complated.

**Which issue(s) this PR fixes**: 
Fixes https://github.com/apache/dubbo-go/issues/2030

**You should pay attention to items below to ensure your pr passes our ci test**
We do not merge pr with ci tests failed

- [ ] All ut passed (run 'go test ./...' in project root)
- [ ] After go-fmt ed , run 'go fmt project' using goland.
- [ ] Golangci-lint passed, run 'sudo golangci-lint run' in project root.
- [ ] After import formatted, (using [imports-formatter](https://github.com/dubbogo/tools#5-how-to-get-imports-formatter) to run 'imports-formatter .' in project root, to format your import blocks, mentioned in [CONTRIBUTING.md](https://github.com/apache/dubbo-go/blob/master/CONTRIBUTING.md) above) 
- [ ] Your new-created file needs to have [apache license](https://raw.githubusercontent.com/dubbogo/resources/master/tools/license/license.txt) at the top, like other existed file does.
- [ ] All integration test passed. You can run integration test locally (with docker env). Clone our [dubbo-go-samples](https://github.com/apache/dubbo-go-samples) project and replace the go.mod to your dubbo-go, and run 'sudo sh start_integration_test.sh' at root of samples project root. (M1 Slice is not Support)

### CODE REVIEW

**Comment by thehackercat** [comment]
@chickenlj I am wondering if it's needed to add a `sync.WaitGroup` after `NotifyAll` to wait notification completed or not.

File: registry/event/service_instances_changed_listener_impl.go [hunk 1]
```diff
@@ -130,12 +125,15 @@ func (lstn *ServiceInstancesChangedListenerImpl) OnEvent(e observer.Event) error
 
 		for key, notifyListener := range lstn.listeners {
 			urls := lstn.serviceUrls[key]
+			events := make([]*registry.ServiceEvent, len(urls))
```
Comment (binbin0325) [inline registry/event/service_instances_changed_listener_impl.go:128]: should use make([]*registry.ServiceEvent, 0 , len(urls))

File: registry/event/service_instances_changed_listener_impl.go [hunk 2]
```diff
@@ -19,20 +19,15 @@ package event
...
-	"github.com/dubbogo/gost/log/logger"
-)
 
-import (
 	"dubbo.apache.org/dubbo-go/v3/common"
 	"dubbo.apache.org/dubbo-go/v3/common/constant"
 	"dubbo.apache.org/dubbo-go/v3/common/extension"
 	"dubbo.apache.org/dubbo-go/v3/registry"
 	"dubbo.apache.org/dubbo-go/v3/remoting"
+	gxset "github.com/dubbogo/gost/container/set"
```
Comment (binbin0325) [inline registry/event/service_instances_changed_listener_impl.go:28]: go import format

**Review by justxuewei — APPROVED** [review]

File: registry/event/service_instances_changed_listener_impl.go (hunk 1 above)
Comment (thehackercat) [reply registry/event/service_instances_changed_listener_impl.go]: updated.

**Comment by thehackercat** [comment]
@justxuewei @chickenlj PTAL~

**Review by binbin0325 — APPROVED** [review]

**Comment by justxuewei** [comment]
> @justxuewei @chickenlj PTAL~

You may execute `go fmt` and `imports-formatter` locally to pass the CI tests.

**Comment by justxuewei** [comment]
> I don't think it's interduced by this PR, should I raise a new PR to fix that?

This error happened a long before, I forgot how to fix it almost. My suggestion is that you could rebase the latest 3.0 branch.

**Comment by thehackercat** [comment]
duplicated with https://github.com/apache/dubbo-go/pull/2043 , closed.

**Pull request closed without merging**